python/ql/lib/semmle/python/security/dataflow/TarSlipCustomizations.qll

```diff
@@ -0,0 +1,149 @@
+/**
+ * Provides default sources, sinks and sanitizers for detecting
+ * "tar slip"
+ * vulnerabilities, as well as extension points for adding your own.
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.Concepts
+private import semmle.python.dataflow.new.BarrierGuards
+private import semmle.python.ApiGraphs
+
+/**
+ * Provides default sources, sinks and sanitizers for detecting
+ * "tar slip"
+ * vulnerabilities, as well as extension points for adding your own.
+ */
+module TarSlip {
+  /**
+   * A data flow source for "tar slip" vulnerabilities.
+   */
+  abstract class Source extends DataFlow::Node { }
+
+  /**
+   * A data flow sink for "tar slip" vulnerabilities.
+   */
+  abstract class Sink extends DataFlow::Node { }
+
+  /**
+   * A sanitizer for "tar slip" vulnerabilities.
+   */
+  abstract class Sanitizer extends DataFlow::Node { }
+
+  /**
+   * DEPRECATED: Use `Sanitizer` instead.
+   *
+   * A sanitizer guard for "tar slip" vulnerabilities.
+   */
+  abstract deprecated class SanitizerGuard extends DataFlow::BarrierGuard { }
+
```

Comment:
Since this is new code, we might as well just not introduce the deprecated class at all.

```suggestion
```